packages/core/src/connectors/index.ts

```diff
@@ -1,39 +1,57 @@
-import { AlipayConnector } from '@logto/connector-alipay';
-import { AlipayNativeConnector } from '@logto/connector-alipay-native';
-import { AliyunDmConnector } from '@logto/connector-aliyun-dm';
-import { AliyunSmsConnector } from '@logto/connector-aliyun-sms';
-import { FacebookConnector } from '@logto/connector-facebook';
-import { GithubConnector } from '@logto/connector-github';
-import { GoogleConnector } from '@logto/connector-google';
-import { SendGridMailConnector } from '@logto/connector-sendgrid-email';
-import { TwilioSmsConnector } from '@logto/connector-twilio-sms';
-import { WeChatConnector } from '@logto/connector-wechat';
-import { WeChatNativeConnector } from '@logto/connector-wechat-native';
+import { existsSync, readFileSync } from 'fs';
+import path from 'path';
+
+import resolvePackagePath from 'resolve-package-path';
 
 import RequestError from '@/errors/RequestError';
 import { findAllConnectors, insertConnector } from '@/queries/connector';
 
+import { connectorPackages } from './consts';
 import { ConnectorInstance, ConnectorType, IConnector, SocialConnectorInstance } from './types';
 import { getConnectorConfig } from './utilities';
 
-const allConnectors: IConnector[] = [
-  new AlipayConnector(getConnectorConfig),
-  new AlipayNativeConnector(getConnectorConfig),
-  new AliyunDmConnector(getConnectorConfig),
-  new AliyunSmsConnector(getConnectorConfig),
-  new FacebookConnector(getConnectorConfig),
-  new GithubConnector(getConnectorConfig),
-  new GoogleConnector(getConnectorConfig),
-  new SendGridMailConnector(getConnectorConfig),
-  new TwilioSmsConnector(getConnectorConfig),
-  new WeChatConnector(getConnectorConfig),
-  new WeChatNativeConnector(getConnectorConfig),
-];
+// eslint-disable-next-line @silverhand/fp/no-let
+let cachedConnectors: IConnector[] | undefined;
```

Comment:
Why do we need to define the variable here? Can it be moved into the `loadConnectors` function?